services/src/functional-tests/src/main/groovy/org/openkilda/functionaltests/extension/spring/SpringContextExtension.groovy

```diff
@@ -37,7 +37,7 @@ class SpringContextExtension extends AbstractGlobalExtension implements Applicat
                 TagExtension.collectAllTags(nonSpecialFeatures[0]).contains(Tag.HARDWARE)
         specInfo.getAllFeatures().find {
             it.featureMethod.getAnnotation(PrepareSpringContextDummy)
-        }?.excluded = !(nonSpecialFeatures[0].parameterized || isProfileDependent) ||
+        }?.excluded = !((nonSpecialFeatures.size() > 0 && nonSpecialFeatures[0].parameterized)|| isProfileDependent) ||
```

Comment:
you can use safe navigation operator `!(nonSpecialFeatures[0]?.parameterized || isProfileDependent)`